libzmix/src/signatures/bbs/pok_sig.rs

```diff
@@ -360,4 +377,80 @@ mod tests {
             .verify(&verkey, revealed_msgs.clone(), &challenge_verifier)
             .unwrap());
     }
+
+    #[test]
+    fn test_PoK_multiple_sigs_with_same_msg() {
+        // Prove knowledge of multiple signatures and the equality of a specific message under both signatures.
+        // Knowledge of 2 signatures and their corresponding messages is being proven.
+        // 2nd message in the 1st signature and 5th message in the 2nd signature are to be proven equal without revealing them
+
+        let message_count = 5;
+        let (vk, signkey) = generate(message_count).unwrap();
+
+        let same_msg = FieldElement::random();
+        let mut msgs_1 = FieldElementVector::random(message_count - 1);
+        msgs_1.insert(1, same_msg.clone());
+        let sig_1 = Signature::new(msgs_1.as_slice(), &signkey, &vk).unwrap();
+        assert!(sig_1.verify(msgs_1.as_slice(), &vk).unwrap());
+
+        let mut msgs_2 = FieldElementVector::random(message_count - 1);
+        msgs_2.insert(4, same_msg.clone());
+        let sig_2 = Signature::new(msgs_2.as_slice(), &signkey, &vk).unwrap();
+        assert!(sig_2.verify(msgs_2.as_slice(), &vk).unwrap());
+
+        // A particular message is same
+        assert_eq!(msgs_1[1], msgs_2[4]);
+
+        let same_blinding = FieldElement::random();
+
+        let mut blindings_1 = FieldElementVector::random(message_count - 1);
+        blindings_1.insert(1, same_blinding.clone());
+
+        let mut blindings_2 = FieldElementVector::random(message_count - 1);
+        blindings_2.insert(4, same_blinding.clone());
+
+        // Blinding for the same message is kept same
+        assert_eq!(blindings_1[1], blindings_2[4]);
```

Comment:
Do the blindings need to be the same in order for the Schnorr proof to work?  I thought it was sufficient for the holder to know the blindings for each hidden value used in the equality proof.

Reply:
The blindings need to be same when proving equality. The response for each `message` is of the form `blinding + challenge*message` where `challenge` is common across all messages (and statements). The verifier expects the response for same messages to be same.

Reply:
How does that work with link secrets, which are blinded differently in each credential? I feel like I'm missing something obvious.